core/domain/wipeout_service.py

```diff
@@ -109,11 +107,103 @@ def _delete_user_models(user_id):
     """
     for model_class in models.Registry.get_storage_model_classes(
             [models.NAMES.user]):
-        if (model_class.get_deletion_policy() !=
-                base_models.DELETION_POLICY.KEEP):
+        if (model_class.get_deletion_policy() not in
+                [base_models.DELETION_POLICY.KEEP,
+                 base_models.DELETION_POLICY.NOT_APPLICABLE]):
             model_class.apply_deletion_policy(user_id)
 
 
+def _hard_delete_explorations_and_collections(pending_deletion_model):
+    """Hard delete the exploration and collection models that are private and
+    solely owned by the user.
+
+    Args:
+        pending_deletion_model: PendingDeletionRequestModel.
+    """
+    exp_services.delete_explorations(
+        pending_deletion_model.id, pending_deletion_model.exploration_ids)
+    collection_services.delete_collections(
+        pending_deletion_model.id, pending_deletion_model.collection_ids)
+
+
+def _generate_activity_to_user_ids_mapping(activity_ids):
+    """Generate mapping from activity IDs to user IDs.
+
+    Args:
+        activity_ids: list(str). List of activity IDs for which to generate
+        new user IDs.
+
+    Returns:
+        dict(str, str). Mapping between the activity IDs and pseudonymized user
+        IDs.
+    """
+    return {
+        activity_id: user_models.PseudonymizedUserModel.get_new_id()
+        for activity_id in activity_ids
+    }
+
+
+def _delete_story_models(pending_deletion_model):
+    """Pseudonymize the story models for the user with user_id.
+
+    Args:
+        pending_deletion_model: PendingDeletionRequestModel.
+    """
+    user_id = pending_deletion_model.id
+    metadata_models = story_models.StorySnapshotMetadataModel.query(
+        story_models.StorySnapshotMetadataModel.committer_id == user_id
+    ).fetch()
+    story_ids = set([
+        model.get_unversioned_instance_id() for model in metadata_models])
+    commit_log_models = story_models.StoryCommitLogEntryModel.query(
+        story_models.StoryCommitLogEntryModel.user_id == user_id
+    ).fetch()
+    story_ids = story_ids | set(model.story_id for model in commit_log_models)
+
+    if not pending_deletion_model.story_mappings:
+        pending_deletion_model.story_mappings = (
+            _generate_activity_to_user_ids_mapping(story_ids))
+        pending_deletion_model.put()
+
+    def _pseudonymize_models(story_related_models, user_id):
```

Comment:
Make the user_id variable name more specific to indicate that it's not a regular user ID -- e.g. pseudonymized_user_id.